src/Umbraco.Cms.Integrations.Search.Algolia/Services/AlgoliaSearchPropertyIndexValueFactory.cs

```diff
@@ -34,7 +34,7 @@ public AlgoliaSearchPropertyIndexValueFactory(IDataTypeService dataTypeService,
 
             var indexValues = dataType.Editor.PropertyIndexValueFactory.GetIndexValues(property, culture, string.Empty, true);
 
-            if(indexValues == null) return default;
+            if (indexValues == null || !indexValues.Any()) return new KeyValuePair<string, string>(property.Alias, string.Empty);
 
             var indexValue = indexValues.FirstOrDefault();
```

Comment:
```suggestion
            var indexValue = indexValues.First();
```

Just be be more explicit that you have already checked there is a value and you won't be returning a default/null value.